compiler/utils.go

```diff
@@ -650,5 +650,8 @@ func (s varsByName) Swap(i, j int) {
 }
 
 func (s varsByName) Less(i, j int) bool {
```

Comment:
`varsByName` is being used in exactly one place in this package. We're requiring Go 1.8, that means we have [`sort.Slice`](https://godoc.org/sort#Slice) available to us. Let's use it.

It will bring this code closer to where it's being used, making it more readable in the relevant context.